lib/OpenQA/CLI/api.pm

```diff
@@ -105,26 +106,34 @@ sub command {
     openqa-cli api -X POST job_templates_scheduling/1 \
       schema=JobTemplates-01.yaml preview=0 template="$(cat foo.yaml)"
 
+    # Post job template (from file)
+    openqa-cli api -X POST job_templates_scheduling/1 \
+      schema=JobTemplates-01.yaml preview=0 --param-file template=foo.yaml
+
     # Post job template (from JSON file)
     openqa-cli api --data-file form.json -X POST job_templates_scheduling/1
 
   Options:
-        --apibase <path>        API base, defaults to /api/v1
-        --apikey <key>          API key
-        --apisecret <secret>    API secret
-    -a, --header <name:value>   One or more additional HTTP headers
-    -D, --data-file <path>      Load content to send with request from file
-    -d, --data <string>         Content to send with request, alternatively you
-                                can also pipe data to openqa-cli
-    -f, --form                  Turn JSON object into form parameters
-        --host <host>           Target host, defaults to http://localhost
-    -h, --help                  Show this summary of available options
-    -j, --json                  Request content is JSON
-        --osd                   Set target host to http://openqa.suse.de
-        --o3                    Set target host to https://openqa.opensuse.org
-    -p, --pretty                Pretty print JSON content
-    -q, --quiet                 Do not print error messages to STDERR
-    -X, --method <method>       HTTP method to use, defaults to GET
-    -v, --verbose               Print HTTP response headers
+        --apibase <path>          API base, defaults to /api/v1
+        --apikey <key>            API key
+        --apisecret <secret>      API secret
+    -a, --header <name:value>     One or more additional HTTP headers
+    -D, --data-file <path>        Load content to send with request from file
+    -d, --data <string>           Content to send with request, alternatively
+                                  you can also pipe data to openqa-cli
+    -f, --form                    Turn JSON object into form parameters
+        --host <host>             Target host, defaults to http://localhost
+    -h, --help                    Show this summary of available options
+    -j, --json                    Request content is JSON
+        --osd                     Set target host to http://openqa.suse.de
+        --o3                      Set target host to https://openqa.opensuse.org
+        --param-file <param=file> Load content of params from files instead of
```

Comment:
In consistency with `--data-file <path>`:
```suggestion
        --param-file <path> Load content of params from files instead of
```

Reply:
But we need the parameter name before the path

Reply:
Use cases are different for both. `--data-file` takes one file and only one file, typically in JSON format, with a list of parameters and its values; in conjunction with `--form` the contents of this file take precedence over parameters passed on the command line itself.

For the examples I've used above, an example of a data file would be:
```
{
    "schema" : "JobTemplates-01.yaml",
    "preview" : "1"
}
```
And not sure if we could add a `template` value there .... in my tests JSON and YAML together in a file did not mix too well (or at all), so job template probably needs to be translated from YAML to JSON in order to include it in the data file, which defeats the purpose of having the template in YAML in the first place.

`--param-file` OTOH can be used several times to read several parameters from several files. On the above examples, think of: `--param-file schema=myschema.txt --param-file preview=mypreview.txt --param-file template=foo.yml` with `myschema.txt` and `mypreview.txt` containing `JobTemplates-01.yaml` and `1` respectively. Of course, it would not make sense to have files for parameters with short values, but it does for something such as the YAML template.

Hope this clears it up.

Reply:
Sorry, I've missed that.